core/src/main/java/org/apache/calcite/sql/dialect/BigQuerySqlDialect.java

```diff
@@ -1621,7 +1621,8 @@ private static void unparseTrim(SqlWriter writer, SqlCall call, int leftPrec,
     // If the trimmed character is a non-space character, add it to the target SQL.
     // eg: TRIM(BOTH 'A' from 'ABCD'
     // Output Query: TRIM('ABC', 'A')
-    String value = requireNonNull(valueToTrim.toValue(), "valueToTrim.toValue()");
+    String value = requireNonNull(Util.removeLeadingAndTrailingSingleQuotes(valueToTrim.toString()),
+        "valueToTrim.toValue()");
```

Comment:
change the message to "valueToTrim.toString()"

Reply:
done